### PR TITLE
Fixes #8205: Add a cfpromises check to rudder agent health

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/8205

### CODE REVIEW

File: share/commands/agent-health [hunk 1]
```diff
@@ -58,5 +58,32 @@ then
...
+/var/rudder/cfengine-community/bin/cf-promises -f failsafe.cf > /dev/null 2>&1
+if [ $? -ne 0 ]
+then
+  echo "Broken failsafe promises"
+  [ -z "${NRPE}" ] && echo "The failsafe promises are broken, you can run 'rudder agent reset' to go back to the initial promises"
+  exit 2 # error
+fi
+
+# test promises
+CFPROMISES=$(/var/rudder/cfengine-community/bin/cf-promises -I 2>&1)
```
Comment (peckpeck) [inline share/commands/agent-health]: Why the -I ?

File: share/commands/agent-health [hunk 2]
```diff
@@ -58,5 +58,32 @@ then
...
+if [ $? -ne 0 ]
+then
+  echo "Broken promises"
+  [ -z "${NRPE}" ] && echo "The promises are broken, you should fix them on the policy server or run 'rudder agent reset' to go back to the initial promises"
+  exit 2 # error
+fi
+
+# test hostname resolution errors
+RESOLUTION=$(printf '%s' "$CFPROMISES" | grep "Unable to lookup hostname")
```
Comment (peckpeck) [inline share/commands/agent-health]: You should use the ${} form

Comment (peckpeck) [inline share/commands/agent-health]: Shouldn't we check resolution error on cf-serverd too ?

File: share/commands/agent-health (hunk 1 above)
Comment (amousset) [inline share/commands/agent-health]: To get the list of not resolving hostnames. Info output of cf-promises is very short .

File: share/commands/agent-health (hunk 2 above)
Comment (amousset) [inline share/commands/agent-health]: It is checked by the cf-promises command (the access rules are loaded in promises.cf).

**Comment by amousset** [comment]
PR updated

**Comment by peckpeck** [comment]
Good, merging

**Pull request merged**